oak_functions/sdk/oak_functions/src/lib.rs

```diff
@@ -94,18 +94,26 @@ pub fn report_metric<T: AsRef<str>>(
     let label = label.as_ref().to_owned();
     let request = ReportMetricRequest { label, value };
 
-    let serialized_request =
-        bincode::serialize(&request).expect("Fail to serialize report metric request.");
-
-    let response = invoke(
-        oak_functions_abi::ExtensionHandle::MetricsHandle,
-        &serialized_request,
-    )?;
-
-    let response: ReportMetricResponse =
-        bincode::deserialize(&response).expect("Failed to deserialize report metric response.");
-
-    Ok(response.result)
+    let response = match bincode::serialize(&request) {
+        Ok(serialized_request) => invoke(
+            oak_functions_abi::ExtensionHandle::MetricsHandle,
+            &serialized_request,
+        ),
+        Err(err) => {
+            log!("Failed to serialize request: {}", err);
+            return Ok(Err(ReportMetricError::DeOrSerializingFailed));
```

Comment:
I find this a bit surprising here that if we cannot serialize the request, we make it look like the request went through and we got this response back. IMO that obfuscates what actually is going on, i.e. that the request was not even sent at all. Perhaps it should return ERR_INVALID_ARGS at the ABI level? Also, thinking how this would work in other extensions: it would not work to figure out which error variant of each extension corresponds more closely to something about serialization; and in a hypothetical IDL, the serialization logic on the client side will have to be auto generated, so we definitely cannot assume that there is a mapping. FWIW the response of the invocation may not even be a Result at all, in general, so we need to convey the serialization error in some other way.